vault.gemspec

```diff
@@ -19,6 +19,8 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
+  spec.add_dependency "net-http-persistent", "~> 3.0"
```

Comment:
This gem is incredibly stable, with the latest release coming this week but previous to that it had been 2 years since the release. I understand your concern about dependencies, but I really do think that using the official gem is better than trying to vendor it.

That being said, if having it as a dependency is a non-starter, I'll work on vendoring.